GPU/Common/TextureDecoder.h

```diff
@@ -58,7 +58,7 @@ typedef u32 ReliableHashType;
 #endif
 
 // For ARM64, NEON is mandatory, so we also statically link.
-#elif PPSSPP_ARCH(ARM64)
+#elif PPSSPP_ARCH(ARM64) || defined(ARM64)
```

Comment:
This looks weird, if ARM64 is defined, PPSSPP_ARCH(ARM64) should probably be defined...

Reply:
I agree, but seems to not work that way. For example, when ARMv7 is set to ON, the cmake_system_processor still auto detects aarch64. Only way around this that I've found is to use FORCED_CPU.

Reply:
This still feels very wrong... Hm.

Reply:
Yeah ok, fair enough. Should find a way to make this cleaner later.

Reply:
I would argue that these:

```c++
#if (defined(_M_SSE) && defined(_M_X64)) || defined(ARM64)
```

Should instead be fixed to be:

```c++
#if (defined(_M_SSE) && defined(_M_X64)) || PPSSPP_ARCH(ARM64)
```

And no change needed here.  It looks sorta wrong, kinda like if you saw this code:

```c++
if (a == 2 && 2 == a) {
```

Yes, it's possible that `operator ==(a, b)` could != `operator ==(b, a)` but that's awful.  Having `PPSSPP_ARCH(ARM64) || defined(ARM64)` implies that PPSSPP's code has subtle but disgusting gotchas, which I hope it doesn't.

-[Unknown]

Reply:
I believe that change would accomplish the same goal, agreed. :) Question for you related to a potential "gotcha"... in ppsspp_config.h on the conditional segment starting on line 56 for aarch64, you only define ARM64=1 if it's not yet defined. What if the user explicitly defines ARM64=0 but the cmake_system_processor identifies it's aarch64? A user-case would be for a 64-bit kernel running 32-bit userland. Wouldn't that technically result in PPSSPP_ARCH(ARM64) being 1 but ARM64 being 0?

Reply:
`defined(ARM64)` would be true even if `ARM64` is set to 0 or anything else.  It doesn't matter what it's set to.

-[Unknown]